packages/p2p/stores/buy-sell-store.js

```diff
@@ -0,0 +1,360 @@
+import { observable, action } from 'mobx';
+import { localize } from 'Components/i18next';
+import { height_constants } from 'Utils/height_constants';
+import { requestWS } from 'Utils/websocket';
+import { formatMoney, getDecimalPlaces, getRoundedNumber } from '@deriv/shared';
+import { buy_sell } from '../src/constants/buy-sell';
+import { textValidator, lengthValidator } from 'Utils/validations';
+import { countDecimalPlaces } from 'Utils/string';
+
+export default class BuySellStore {
+    constructor(root_store) {
+        this.root_store = root_store;
+        this.general_store = this.root_store.general_store;
+    }
+
+    @observable api_error_message = '';
+    @observable contact_info = '';
+    @observable error_message = '';
+    @observable has_more_items_to_load = false;
+    @observable is_loading = true;
+    @observable is_submit_disabled = true;
+    @observable item_offset = 0;
+    @observable items = [];
+    @observable payment_info = '';
+    @observable receive_amount = 0;
+    @observable selected_ad_state = {};
+    @observable should_show_popup = false;
+    @observable should_show_verification = false;
+    @observable show_advertiser_page = false;
+    @observable submitForm = () => {};
+    @observable table_type = buy_sell.BUY;
+
+    form_props = {};
+    height_values = [
+        height_constants.screen,
+        height_constants.core_header,
+        height_constants.page_overlay_header,
+        height_constants.page_overlay_content_padding,
+        height_constants.tabs,
+        height_constants.filters,
+        height_constants.filters_margin,
+        height_constants.table_header,
+        height_constants.core_footer,
+    ];
+    initial_values = {
+        amount: this.advert?.min_order_amount_limit,
+        // For sell orders we require extra information.
+        ...(this.is_sell_advert ? { contact_info: this.contact_info, payment_info: this.payment_info } : {}),
+    };
+    is_buy = this.table_type === buy_sell.BUY;
+    item_height = 56;
+
+    get account_currency() {
+        return this.advert?.account_currency;
+    }
+
+    get advert() {
+        return this.form_props?.advert;
+    }
+
+    get is_buy_advert() {
+        return this.advert?.counterparty_type === buy_sell.BUY;
+    }
+
+    get is_sell_advert() {
+        return this.advert?.counterparty_type === buy_sell.SELL;
+    }
+
+    get modal_title() {
+        if (this.is_buy_advert) {
+            return localize('Buy {{ account_currency }}', { account_currency: this.account_currency });
+        } else {
+            return localize('Sell {{ account_currency }}', { account_currency: this.account_currency });
+        }
+    }
```

Comment:
Please make all of these `computed`, by doing so you'll instruct MobX to cache these getters, i.e. the value will be re-calculated only when one of the referenced observables changes.